src/main/scala/is/hail/utils/Context.scala

```diff
@@ -6,13 +6,16 @@ abstract class Context {
 
 case class TextContext(line: String, file: String, position: Option[Int]) extends Context {
   def wrapException(e: Throwable): Nothing = {
-    val msg = e match {
-      case _: FatalException => e.getMessage
-      case _ => s"caught $e"
+    e match {
+      case _: HailException =>
+        fatal(
+          s"""$file${ position.map(ln => ":" + (ln + 1)).getOrElse("") }: ${ e.getMessage }
+             |  offending line: @1""".stripMargin, line)
+      case _ => e.getMessage
+        fatal(
+          s"""$file${ position.map(ln => ":" + (ln + 1)).getOrElse("") }: caught ${ e.getClass.getName() }: ${ e.getMessage }
+             |  offending line: @1""".stripMargin, line)
```

Comment:
I know your PR doesn't change this, but it'd be nice if we preserved the stack traces from `e` but making them the `cause` of the `FatalException` (`cause` is inherited from `RuntimeException`).